### DISCUSSION TITLE
Difference between two dates

### DISCUSSION THREAD

**Discussion by manuelnegroprima** [Budibase/budibase#13446] [Help] [answered]
Hi!

I need to create formula that return tu me the number of days between 2 days.
Particularly, one date is in the past (this date is stored in the record) and the other day is the current day.
After, I need to multiply the result with a price.

Anyone know how I can do the difference between the 2 dates?
Thank you in advance!

Manuel

**Comment by mikesealey** [comment, marked as answer]
Hi @manuelnegroprima, great question

For this use-case, I recommend a little bit of JavaScript.

```
// If no row has been clicked, prompt the user
if (!$("State.clickedRow")) {
  return "Please select a row"
}

// Turn the clicked row into JavaScript Object Notation (JSON)
let clickedRow = JSON.parse($("State.clickedRow")) || {}

// Assign the start and end dates
let startDate = new Date(clickedRow.Works_Start)
let endDate = new Date(clickedRow.Works_End)

// Converts to milliseconds since epoch, subtracts one from the other
let differenceMs = endDate.getTime() - startDate.getTime()

// From milliseconds back to days
let millisecondsInADay = 1000 * 60 * 60 * 24
let daysDifference = Math.round(differenceMs / millisecondsInADay)

// And returns the difference in days
return "Days difference: " + daysDifference
```

This code is using the Budibase sample data from the Jobs table to compare two dates.

A few things to consider: 
* This will round to the nearest day, which is possibly zero days difference
* _**You should adapt and test this for your use-case**_

Some further reading might be 
* [Math.ceil](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/ceil) which will always round up.
* [Math.floor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/floor) which will always round down.

I've also attached the app I built the above formula in, so that you can take a closer look at how it's working.
[13446manuelnegroprima-export-1713454010685.tar.gz](https://github.com/Budibase/budibase/files/15027072/13446manuelnegroprima-export-1713454010685.tar.gz)

You can read how to import that app [here, in the Budibase documentation](https://docs.budibase.com/docs/export-and-import-apps#import-an-application).
